### PR TITLE
Functionality added to SES service

### PR DESCRIPTION
**Pull request by pinzon**
This patch adds the functions test_render_template and update_template so the SES service.
Also adds validation to the function add_template.

Each new function has its corresponding test.

This should close issue localstack/localstack/issues/3466

### CODE REVIEW

**Comment by pinzon** [comment]
Hi @bblommers sorry to bother you. Could you review my PR? 
Is there something else I should do? some tests failed but I don't think they have to do with my changes.
Thank you.

File: tests/test_ses/test_ses_boto3.py [hunk 1]
```diff
@@ -484,3 +485,66 @@ def test_create_ses_template():
...
+            "TemplateName": "MyTestTemplate",
+            "SubjectPart": "Greetings, {{name}}!",
+            "TextPart": "Dear {{name}},"
+            "\r\nYour favorite animal is {{favoriteanimal}}.",
+            "HtmlPart": "<h1>Hello {{name}},"
+            "</h1><p>Your favorite animal is {{favoriteanimal}}.</p>",
+        }
+    )
+    result = conn.test_render_template(**kwargs)
+    assert True == result["RenderedTemplate"].__contains__("Subject: Greetings, John!")
```
Comment (bblommers) [inline tests/test_ses/test_ses_boto3.py:512]: Please use the should-flow for these assertions:
`template.should.contain(...)`

File: tests/test_ses/test_ses_boto3.py [hunk 2]
```diff
@@ -484,3 +485,66 @@ def test_create_ses_template():
...
+@mock_ses
+def test_render_template():
+    conn = boto3.client("ses", region_name="us-east-1")
+
+    kwargs = dict(
+        TemplateName="MyTestTemplate",
+        TemplateData=json.dumps({"name": "John", "favoriteanimal": "Lion"}),
+    )
+
+    conn.test_render_template.when.called_with(**kwargs).should.throw(ClientError)
```
Comment (bblommers) [inline tests/test_ses/test_ses_boto3.py:499]: When testing error scenarios, we prefer to use pytest.raises, as that makes it easier to assert that the correct error was thrown:
```
with pytest.raises(ClientError) as ex:
  # something that fails
err  = ex.value.response["Error"]
err["Code"].should.equal("...")
err["Message"].should.equal("...")
```

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks for the PR, @pinzon! I've left a few comments.
Looks like the build is failing as well, because of a Py2 issue: https://github.com/spulec/moto/pull/3670/checks?check_run_id=1872455785

**Comment by pinzon** [comment]
Hi @bblommers, my last commit should fix that error and I changed the tests as requested.
Thank you for your help.

**Review by bblommers — APPROVED** [review]
LGTM. Thanks again @pinzon!

**Pull request merged**